realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java

```diff
@@ -1827,6 +1827,7 @@ public void isValid_removedParent() {
         list.removeAll(SUPPORTED_IS_EMPTY_TYPES);
         list.remove(RealmFieldType.UNSUPPORTED_MIXED);
         list.remove(RealmFieldType.UNSUPPORTED_TABLE);
+        list.remove(RealmFieldType.UNSUPPORTED_DATE);
```

Comment:
The trick is that when you add a new type, your IDE will be unhappy in all the `switch` constructors so you remember to update them all.